.github/workflows/main.yml

```diff
@@ -0,0 +1,375 @@
+#          md -Force ${{github.workspace}}\Libraries\
+#          if (!(Test-Path -Path "${{github.workspace}}\Libraries\TIXI-3.1.1-win64")) {
+#            Write-Output "Downloading TiXI libraries";
+#            Invoke-WebRequest -Uri "https://github.com/DLR-SC/tixi/releases/download/v3.1.1/TIXI-3.1.1-win64.zip" -OutFile "${{github.workspace}}\Libraries\TIXI-3.1.1-win64.zip";
+#            Write-Output "Extract TiXI libraries";
+#            7z x -y "${{github.workspace}}\Libraries\TIXI-3.1.1-win64.zip" -o${{github.workspace}}\Libraries
+#          }
+#      - name: Install ninja
+#        run: choco install -y ninja
+#      - name: Build
+#        run: |
+#          cd cpp
+#          md -Force build-win
+#          cd build-win
+#          cmake --version
+#          cmake -G "Ninja" -DCMAKE_BUILD_TYPE=Release -DEPI_USE_BUNDLED_EIGEN=ON -DEPI_BUILD_EPI_IO=ON -DHDF5_USE_STATIC_LIBRARIES=ON -DHDF5_DIR=${{github.workspace}}\Libraries\HDF5-1.12.0-win64\cmake\hdf5 -Dtixi3_DIR=${{github.workspace}}\Libraries\TIXI-3.1.1-win64\lib\tixi3 ..
+#          cp "${{github.workspace}}\Libraries\TIXI-3.1.1-win64\bin\tixi3.dll" "tests"
+#          cmake --build .
+#      - name: Upload built test directory
+#        uses: actions/upload-artifact@v2
+#        with:
+#          name: build-tests-windows
+#          path: cpp/build-win/tests
+#          retention-days: 1
+
+
+  test-cpp:
+    needs: build-cpp
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install Dependencies
+        run: |
+          sudo apt-get -qq update
+          sudo apt-get -qq -y install libhdf5-cpp-100 wget gnupg
+          echo 'deb http://download.opensuse.org/repositories/science:/dlr/xUbuntu_18.04/ /' | sudo tee /etc/apt/sources.list.d/science:dlr.list
+          wget -O - https://download.opensuse.org/repositories/science:dlr/xUbuntu_18.04/Release.key  | sudo apt-key add -
+          sudo apt-get -qq update
+          sudo apt-get -qq -y install libtixi3 g++ python3-pip
+          python3 -m pip install gcovr
+      - name: Download built test directory
+        uses: actions/download-artifact@v2
+        with:
+          name: build-tests-linux
+          path: cpp/build/tests
+      - name: Run unit tests
+        run: |
+          cd cpp/build/tests
+          sudo chmod a+x runUnitTests
+          ./runUnitTests  --gtest_output="xml:report.xml"
+          cd ../../../
+          gcovr -f cpp/epidemiology
+          gcovr -f cpp/epidemiology --xml -o coverage.xml
+          mkdir coverage_report && cd coverage_report
+          gcovr -r .. -f ../cpp/epidemiology --exclude-throw-branches --html --html-details -o index.html
+      - name: Upload coverage reports
+        uses: actions/upload-artifact@v2
+        with:
+          name: test-cpp-coverage-reports
+          path: |
+            cpp/build/tests/report.xml
+            coverage.xml
+            coverage_report
+          retention-days: 1
+
+#  test-cpp-win:
```

Comment:
should be reenabled

Reply:
Finally! :goberserk: @mknaranja, could you have another look?

BTW, I also figured out why the C++ coverage was missing from codecov and fixed that.